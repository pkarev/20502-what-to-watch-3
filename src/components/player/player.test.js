import React from 'react';
import renderer from 'react-test-renderer';
import Player from './player.jsx';

const movie = {
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  posterBig: `img/bg-the-grand-budapest-hotel.jpg`,
};

it(`Render Player`, () => {
  const tree = renderer
    .create(
        <Player movie={movie} onPlayerExitClick={() => {}}/>, {
          createNodeMock: () => ({}),
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
