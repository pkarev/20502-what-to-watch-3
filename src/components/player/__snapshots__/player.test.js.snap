// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Player 1`] = `
<div
  className="player"
  onClick={[Function]}
>
  <video
    autoPlay={true}
    controls={false}
    muted={false}
    poster="img/bg-the-grand-budapest-hotel.jpg"
    style={
      Object {
        "height": "100%",
        "objectFit": "cover",
        "objectPosition": "center center",
        "width": "100%",
      }
    }
  >
    <source
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
  </video>
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="0"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        --:--
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
